libp2p/transports/quictransport.nim

```diff
@@ -0,0 +1,242 @@
+import std/sequtils
+import pkg/chronos
+import pkg/chronicles
+import pkg/quic
+import ../multiaddress
+import ../multicodec
+import ../stream/connection
+import ../wire
+import ../muxers/muxer
+import ../upgrademngrs/upgrade
+import ./transport
+
+export multiaddress
+export multicodec
+export connection
+export transport
+
+logScope:
+  topics = "libp2p quictransport"
+
+type
+  P2PConnection = connection.Connection
+  QuicConnection = quic.Connection
+
+# Stream
+type
+  QuicStream* = ref object of P2PConnection
+    stream: Stream
+    cached: seq[byte]
+
+proc new(_: type QuicStream, stream: Stream, oaddr: MultiAddress, peerId: PeerId): QuicStream =
+  let quicstream = QuicStream(stream: stream, observedAddr: oaddr, peerId: peerId)
+  procCall P2PConnection(quicstream).initStream()
+  quicstream
+
+template mapExceptions(body: untyped) =
+  try:
+    body
+  except QuicError:
+    raise newLPStreamEOFError()
+
+method readOnce*(stream: QuicStream,
+                 pbytes: pointer,
+                 nbytes: int): Future[int] {.async.} =
+  if stream.cached.len == 0:
+    stream.cached = await mapExceptions(stream.stream.read())
+  if stream.cached.len <= nbytes:
```

Comment:
`readOnce` is best effort, it reads up to `nbytes` and returns as much as it could read. We should simply return whatever `stream.stream.read()` returns...